bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/handler/ZwaveJSNodeHandler.java

```diff
@@ -569,40 +620,33 @@ public void dispose() {
-        } else if (commandClassIds.removeAll(COMMAND_SET_HVAC_EQUIPMENT)) {
-            return Equipment.HVAC;
-        } else if (commandClassIds.removeAll(COMMAND_SET_IRRIGATION_EQUIPMENT)) {
-            return Equipment.IRRIGATION;
-        } else if (commandClassIds.removeAll(COMMAND_SET_LOCK_EQUIPMENT)) {
-            return Equipment.LOCK;
-        } else if (commandClassIds.removeAll(COMMAND_SET_METER_EQUIPMENT)) {
-            return Equipment.ELECTRIC_METER;
-        } else if (commandClassIds.removeAll(COMMAND_SET_POWER_SUPPLY_EQUIPMENT)) {
-            return Equipment.POWER_SUPPLY;
-        } else if (commandClassIds.removeAll(COMMAND_SET_SENSOR_EQUIPMENT)) {
-            return Equipment.SENSOR;
-        } else if (commandClassIds.removeAll(COMMAND_SET_THERMOSTAT_EQUIPMENT)) {
-            return Equipment.THERMOSTAT;
-        } else if (commandClassIds.removeAll(COMMAND_SET_WINDOW_COVERING_EQUIPMENT)) {
-            return Equipment.WINDOW_COVERING;
-        } else if (commandClassIds.removeAll(COMMAND_SET_ZONE_EQUIPMENT)) {
-            return Equipment.ZONE;
+        // Map of command class sets to their corresponding equipment tags
+        Map<Set<Integer>, SemanticTag> equipmentMap = Map.ofEntries(
+                Map.entry(COMMAND_SET_LIGHT_SOURCE_EQUIPMENT, Equipment.LIGHT_SOURCE),
+                Map.entry(COMMAND_SET_ALARM_DEVICE_EQUIPMENT, Equipment.ALARM_DEVICE),
+                Map.entry(COMMAND_SET_AUDIO_VISUAL_EQUIPMENT, Equipment.AUDIO_VISUAL),
+                Map.entry(COMMAND_SET_BATTERY_EQUIPMENT, Equipment.BATTERY),
+                Map.entry(COMMAND_SET_CONTROL_DEVICE_EQUIPMENT, Equipment.CONTROL_DEVICE),
+                Map.entry(COMMAND_SET_DISPLAY_EQUIPMENT, Equipment.DISPLAY),
+                Map.entry(COMMAND_SET_GATE_EQUIPMENT, Equipment.GATE),
+                Map.entry(COMMAND_SET_HUMIDIFIER_EQUIPMENT, Equipment.HUMIDIFIER),
+                Map.entry(COMMAND_SET_HVAC_EQUIPMENT, Equipment.HVAC),
+                Map.entry(COMMAND_SET_IRRIGATION_EQUIPMENT, Equipment.IRRIGATION),
+                Map.entry(COMMAND_SET_LOCK_EQUIPMENT, Equipment.LOCK),
+                Map.entry(COMMAND_SET_METER_EQUIPMENT, Equipment.ELECTRIC_METER),
+                Map.entry(COMMAND_SET_POWER_SUPPLY_EQUIPMENT, Equipment.POWER_SUPPLY),
+                Map.entry(COMMAND_SET_SENSOR_EQUIPMENT, Equipment.SENSOR),
+                Map.entry(COMMAND_SET_THERMOSTAT_EQUIPMENT, Equipment.THERMOSTAT),
+                Map.entry(COMMAND_SET_WINDOW_COVERING_EQUIPMENT, Equipment.WINDOW_COVERING),
+                Map.entry(COMMAND_SET_ZONE_EQUIPMENT, Equipment.ZONE));
+
+        // Find the first matching equipment tag based on intersection
+        for (Map.Entry<Set<Integer>, SemanticTag> entry : equipmentMap.entrySet()) {
+            Set<Integer> intersection = new java.util.HashSet<>(commandClassIds);
+            intersection.retainAll(entry.getKey());
+            if (!intersection.isEmpty()) {
```

Comment:
doesn't retainAll return a boolean if the intersection was modified?

Reply:
Will come back to this, will need to do some tests.

Reply:
Fixed